mdns.h

```diff
@@ -0,0 +1,1205 @@
+/* mdns.h  -  mDNS/DNS-SD library  -  Public Domain  -  2017 Mattias Jansson
```

Comment:
Since it looks like this is under active development upstream, can we put the git hash that we used?

https://github.com/mjansson/mdns/commits/master/mdns.h